plugin-server/src/utils/ai-cost-data/mappings.ts

```diff
@@ -0,0 +1,234 @@
+import { ModelDetailsMap, ModelRow } from 'types'
+
+import { anthropicProvider } from './anthropic'
+import { costs as awsBedrockCosts } from './awsBedrock'
+import { costs as azureCosts } from './azure'
+import { costs as cohereCosts } from './cohere'
+import { costs as fireworksAICosts } from './fireworks'
+import { costs as googleCosts } from './google'
+import { costs as groqCosts } from './groq'
+import { costs as mistralCosts } from './mistral'
+import { openAIProvider } from './openai'
+import { costs as fineTunedOpenAICosts } from './openai/fine-tuned-models'
+import { costs as openRouterCosts } from './openrouter'
+import { costs as qstashCosts } from './qstash'
+import { costs as togetherAIChatCosts } from './togetherai/chat'
+import { costs as togetherAIChatLlamaCosts } from './togetherai/chat/llama'
+import { costs as togetherAICompletionCosts } from './togetherai/completion'
+import { costs as togetherAICompletionLlamaCosts } from './togetherai/completion'
+
+const openAiPattern = /^https:\/\/api\.openai\.com/
+const anthropicPattern = /^https:\/\/api\.anthropic\.com/
+const azurePattern = /^(https?:\/\/)?([^.]*\.)?(openai\.azure\.com|azure-api\.net)(\/.*)?$/
+const localProxyPattern = /^http:\/\/127\.0\.0\.1:\d+\/v\d+\/?$/
+const heliconeProxyPattern = /^https:\/\/oai\.hconeai\.com/
+const amdbartekPattern = /^https:\/\/.*\.amdbartek\.dev/
+const anyscalePattern = /^https:\/\/api\.endpoints\.anyscale\.com/
+const cloudflareAiGatewayPattern = /^https:\/\/gateway\.ai\.cloudflare\.com/
+const twoYFV = /^https:\/\/api\.2yfv\.com/
+const togetherPattern = /^https:\/\/api\.together\.xyz/
+const lemonFox = /^https:\/\/api\.lemonfox\.ai/
+const fireworks = /^https:\/\/api\.fireworks\.ai/
+const perplexity = /^https:\/\/api\.perplexity\.ai/
+const googleapis = /^https:\/\/(.*\.)?googleapis\.com/
+// openrouter.ai or api.openrouter.ai
+const openRouter = /^https:\/\/(api\.)?openrouter\.ai/
+//api.wisdominanutshell.academy
+const wisdomInANutshell = /^https:\/\/api\.wisdominanutshell\.academy/
+// api.groq.com
+const groq = /^https:\/\/api\.groq\.com/
+// cohere.ai
+const cohere = /^https:\/\/api\.cohere\.ai/
+// api.mistral.ai
+const mistral = /^https:\/\/api\.mistral\.ai/
+// https://api.deepinfra.com
+const deepinfra = /^https:\/\/api\.deepinfra\.com/
+//https://qstash.upstash.io/llm
+const qstash = /^https:\/\/qstash\.upstash\.io/
+//https://www.firecrawl.dev/
+const firecrawl = /^https:\/\/api\.firecrawl\.dev/
+// https://bedrock-runtime.{some-region}.amazonaws.com/{something-after}
+const awsBedrock = /^https:\/\/bedrock-runtime\.[a-z0-9-]+\.amazonaws\.com\/.*/
+
+export const providersNames = [
```

Comment:
Could we autogenerate this from `providers` so that there's only one place to update?